packages/tools/src/types/EventTypes.ts

```diff
@@ -116,8 +118,13 @@ type AnnotationAddedEventDetail = {
  * drawing on the viewport.
  */
 type AnnotationCompletedEventDetail = {
+  /** unique id of the viewport */
+  viewportId?: string;
+  /** unique id of the rendering engine */
+  renderingEngineId?: string;
```

Comment:
actually we should go the other way around and remove the viewportId, and renderingEngineId from the other ones. I really hated that annotationAdded and modified are emitting viewport/renderingEngineId

Reply:
How do you want that implemented then?  The current interpolation code depends on having those values.  Is this a change we can make right now given that there might be others depending on having those values?